### PR TITLE
Feature/docsite migration

### PR DESCRIPTION
**Pull request by mahsa-ameri**
- [x] Moved these components from old doc site: Side-Nav, title-slug-url, overview, for-designers, for-developers, Utilities, Buttons(basic component)
- [x] Updated routes for all of them
- [x] Added new font link to index.html to support font of code block
- [x] Updated title slug URL to include header 1 and 2(which is with h3 style)
- [x] Updated and match all page's styles based on Figma
- [x] Removed all !important tags (except in grid.scss that is gonna be replace with new grid system)
- [x] Added core to component library package-json **(run `npm i` from component-library)**
- [x] Updated translations
- [x] Added encapsulated component for styling elements within Translation file (like: adding class:'link' to a tags in en and fr.json files would add the style)

### CODE REVIEW

File: component-library/src/app/components/info-text-small/info-text-small.component.html [hunk 1]
```diff
@@ -1,4 +1,5 @@
 <!-- <jds-font-icon [iconConfig]="circleInfoIcon"></jds-font-icon> -->
```
Comment (xingruoxicic) [inline component-library/src/app/components/info-text-small/info-text-small.component.html:1]:
```suggestion
```

File: component-library/src/app/components/side-nav/side-nav.component.html [hunk 2]
```diff
@@ -0,0 +1,40 @@
+<!-- <jds-font-icon -->
```
Comment (xingruoxicic) [inline component-library/src/app/components/side-nav/side-nav.component.html:1]:
```suggestion
```

File: component-library/src/app/components/side-nav/side-nav.component.scss [hunk 3]
```diff
@@ -0,0 +1,111 @@
+@use 'src/scss/grid';
+@use "../../../../../core-library/tokens/sizes" as ircc-ds-sizes;
+
+$footer-height: 240px;
+
+:host {
+  // position: fixed;
```
Comment (xingruoxicic) [inline component-library/src/app/components/side-nav/side-nav.component.scss:7]:
```suggestion
```

**Review by xingruoxicic — APPROVED** [review]

File: component-library/src/app/pages/overview/overview.component.html [hunk 4]
```diff
@@ -3,20 +3,22 @@
     [config]="overViewSlugTitleURLConfig"
   ></app-title-slug-url>
   <h2
-    class="sub-title emphasis"
+    class="sub-title"
     [style.margin-top]="0"
   >
-    {{ 'Overview.SubHeading' | translate }}
+    <strong>{{ 'Overview.SubHeading' | translate }}</strong>
```
Comment (Naseer-007) [inline component-library/src/app/pages/overview/overview.component.html:9]: is the strong tag preferred over emphasis?

File: component-library/src/app/shell/shell.component.html [hunk 5]
```diff
@@ -1,10 +1,29 @@
...
+    </div>
   </div>
   <div class="main-container top {{ navStatus }}">
-    <div class="app-side-nav left-nav">
+    <app-side-nav
+      class="left-nav"
+      [navBarData]="leftNavData"
+      [mobileToggleIcon]="true"
+    ></app-side-nav>
+    <!-- <div class="app-side-nav left-nav">
```
Comment (Naseer-007) [inline component-library/src/app/shell/shell.component.html:26]: Can we remove the unused comments?

File: component-library/src/app/shell/shell.component.scss [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+        font-size: 30px;
+      }
+    }
+  }
+}
+
+.main-container {
+  &.nav-open {
+    display: flex;
+    /** These 3 lines doesn't do anything **/
```
Comment (Naseer-007) [inline component-library/src/app/shell/shell.component.scss:43]: 😆

File: documentation/components/TitleSlugUrlComponent.html [hunk 7]
```diff
@@ -260,7 +260,6 @@ <h3 id="inputs">Inputs</h3>
                 <tr>
                     <td class="col-md-4">
                         <i>Default value : </i><code>{
-    type: slugTitleURLType.primary,
     title: &#x27;&#x27;,
```
Comment (Naseer-007) [inline documentation/components/TitleSlugUrlComponent.html:263]: What does this set?

File: component-library/src/app/pages/button/button.component.html
[the remark this answers is not in the document]
Comment (Naseer-007) [reply component-library/src/app/pages/button/button.component.html:1]: yea we already have one called -button-documentation

**Review by Naseer-007 — APPROVED** [review]

**Pull request merged**